Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1220,18 +1222,20 @@ private boolean readAndStripParam(Uri data, Activity activity) {
             // Capture the intent URI and extra for analytics in case started by external intents such as  google app search
             try {
                 if (data != null) {
+                    boolean foundSchemeMatch = false;
+                    boolean skipThisPath = false;
                     if (externalUriWhiteList_.size() > 0) {
-                        String externalIntentScheme = Uri.parse(data.toString()).getScheme();
-                        if (externalIntentScheme != null && externalUriWhiteList_.contains(externalIntentScheme)) {
-                            sessionReferredLink_ = data.toString();
-                            prefHelper_.setExternalIntentUri(data.toString());
-                        }
-                    } else {
+                        foundSchemeMatch = externalUriWhiteList_.contains(data.getScheme());
+                    }
+                    if (skipExternalUriPaths_.size() > 0) {
+                        skipThisPath = skipExternalUriPaths_.contains(data.getPath());
```

Comment:
Gonna have to check the black-listed path against the beginning of the URI path. So a black-listed path of "products" should blacklist "app://products" as well as "app://products/1234". Right now this only blocks the former.